tools/@aws-cdk/spec2cdk/lib/cdk/resource-decider.ts

```diff
@@ -221,7 +222,7 @@ export class ResourceDecider {
             expr.object({ tagPropertyName: expr.lit(originalName) }),
           ),
         cfnValueToRender: {
-          [originalName]: $this.tags.renderTags(...(HAS_25610 ? [$this[originalName]] : [])),
```

Comment:
Why this change?

Reply:
This is necessary because the `tags` type is `public tags?: Array<cdk.CfnTag>;` instead of `TagManager`. 

So the original code `this.tags.renderTags(...)` won't work here since `tags` is an array which doesn't have `renderTags` method. However, since it inherits `ITaggableV2`, it has direct access to `cdkTagManager` to access `renderTags`.

Reply:
We still need to pass the directly-passed tags in though, to combine them with the aspect-applied tags?

Should this not look like:

```ts
 [originalName]: $this.cdkTagManager.renderTags($this[originalName]),
```

Also, since we effectively seem to have removed all reference to `HAS_25610`, we can as well remove the boolean constant I think?

Reply:
Good point. Will update

Reply:
The reason that `$this[originalName]` is not needed is because I provided initial tag in the initizer. My understanding is that both serves the same purpose. Correct me if wrong.
```
initializer: (props: Expression) =>
          new CDK_CORE.TagManager(
            this.tagManagerVariant(variant),
            expr.lit(this.resource.cloudFormationType),
            $E(props)[originalName],
            expr.object({ tagPropertyName: expr.lit(originalName) }),
          ),
```